contracts/subdaos/cwd-subdao-core/src/contract.rs

```diff
@@ -660,11 +670,20 @@ pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractE
 /// Validates the sender permissions to execute contract's messages. Valid senders are timelock
 /// contracts behind proposal modules of the DAO.
 pub(crate) fn execution_access_check(deps: Deps, sender: Addr) -> Result<(), ContractError> {
+    let res = proposal_from_timelock(deps, sender.to_string())?;
+    match res {
+        Some(_) => Ok(()),
+        None => Err(ContractError::Unauthorized {}),
+    }
+}
+
+// returns Unauthorized if not found
+fn proposal_from_timelock(deps: Deps, sender: String) -> Result<Option<ProposalModule>, StdError> {
```

Comment:
Please add a doc comment for this method what does exactly this method do (what is the purpose of it and what does it return. `Result<Option<ProposalModule>, StdError>` is quite scary, do we really need this?).

Also `returns Unauthorized if not found` comment is wrong, it doesn't return such error